src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const searchKey = String(item[key]);
+
+    if (!result[searchKey]) {
+      result[searchKey] = [];
+    }
+
+    result[searchKey].push(item);
+  });
```

Comment:
Consider using the array reduce method instead of forEach for a more functional approach. The reduce method can help to directly construct the result object without mutating an external variable, which can make the code easier to understand and maintain. However, this is more of a stylistic choice and the current implementation is still correct.